index.js

```diff
@@ -1,924 +1,578 @@
-  var self = this
 
-  this.tree.ready(function (err) {
-    if (err) return cb(err)
-    self.metadata.ready(function (err) {
+  _del (name, cb) {
+    this.ready(err => {
       if (err) return cb(err)
-      if (self.content) return cb(null)
+      this._db.del(name, (err, node) => {
+        if (err) return cb(err)
+        if (!node) return cb(new errors.FileNotFound(name))
+        return cb(null)
+      })
+    })
+  }
 
-      self.key = self.metadata.key
-      self.discoveryKey = self.metadata.discoveryKey
+  unlink (name, cb) {
+    name = unixify(name)
+    this._del(name, cb || noop)
+  }
 
-      if (!self.metadata.writable || self._checkout) onnotwriteable()
-      else onwritable()
-    })
-  })
+  rmdir (name, cb) {
+    if (!cb) cb = noop
+    name = unixify(name)
 
-  function onnotwriteable () {
-    if (self.metadata.has(0)) return self._loadIndex(cb)
-    self._loadIndex(noop)
-    cb()
+    let stream = this._db.iterator(name)
+    stream.next((err, val) => {
+      if (err) return cb(err)
+      if (val) return cb(new errors.DirectoryNotEmpty(name))
+      self._del(name, cb)
+    })
   }
 
-  function onwritable () {
-    var wroteIndex = self.metadata.has(0)
-    if (wroteIndex) return self._loadIndex(cb)
+  replicate (opts) {
+    if (!opts) opts = {}
+    opts.expectedFeeds = 2
 
-    if (!self.content) {
-      var keyPair = contentKeyPair(self.metadata.secretKey)
-      var opts = contentOptions(self, keyPair.secretKey)
-      self.content = hypercore(self._storages.content, keyPair.publicKey, opts)
-      self.content.on('error', function (err) {
-        self.emit('error', err)
-      })
-    }
+    const stream = this.metadataFeed.replicate(opts)
 
-    self.content.ready(function () {
-      if (self.metadata.has(0)) return cb(new Error('Index already written'))
-      self.metadata.append(messages.Index.encode({type: 'hyperdrive', content: self.content.key}), cb)
+    this.contentReady(err => {
+      if (err) return stream.destroy(err)
+      if (stream.destroyed) return
+      this.contentFeed.replicate({
+        live: opts.live,
+        download: opts.download,
+        upload: opts.upload,
+        stream: stream
+      })
     })
-  }
-}
 
-function contentOptions (self, secretKey) {
-  return {
-    sparse: self.sparse || self.latest,
-    maxRequests: self.maxRequests,
-    secretKey: secretKey,
-    storeSecretKey: false,
-    indexing: self.metadata.writable && self.indexing,
-    storageCacheSize: self.contentStorageCacheSize
+    return stream
   }
-}
 
-function isObject (val) {
-  return !!val && typeof val !== 'string' && !Buffer.isBuffer(val)
-}
-
-function wrap (self, storage) {
-  return {
-    metadata: function (name, opts) {
-      return storage.metadata(name, opts, self)
-    },
-    content: function (name, opts) {
-      return storage.content(name, opts, self)
+  checkout (version, opts) {
+    const versionedTrie = this._db.checkout(version) 
```

Comment:
Why not call it `db` and use it directly in the options.

Reply:
Yep, was a leftover from debugging, but now it's removed. Thanks